thesis-api-v2/src/main/java/pl/edu/agh/quizzesthesis/business/service/AnswerService.java

```diff
@@ -0,0 +1,21 @@
+package pl.edu.agh.quizzesthesis.business.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pl.edu.agh.quizzesthesis.api.dto.AnswerRequest;
+import pl.edu.agh.quizzesthesis.business.UserAuthDetails;
+
+@Service
+@AllArgsConstructor
+public class AnswerService {
+
+    private final TermService termService;
+    private final UserService userService;
+
+    @Transactional
+    public void answerQuestion(UserAuthDetails userAuthDetails, AnswerRequest answerRequest) {
+        var term = termService.updateTermDifficulty(answerRequest.termId(), answerRequest.answerCorrect());
+        userService.updateUserRank(userAuthDetails, term, answerRequest.answerCorrect());
```

Comment:
How about renaming it to `updateUserRankInCategory` and passing `userAuthDetails`, `category` and `isAnswerCorrect`. Reason: you are passing whole term when in fact you only needs a category and adding that `InCategory` will explicitly tell that we are updating user ranking in only one category. Also, how about introducing a variable `isAnswerCorrect` and replacing `answerRequest.answerCorrect()`?

Reply:
Done